lib/dynamoid/persistence.rb

```diff
@@ -119,18 +119,7 @@ def undump_field(value, options)
               when :set
                 Set.new(value)
               when :datetime
-                if value.is_a?(Date) || value.is_a?(DateTime) || value.is_a?(Time)
-                  value
-                else
-                  case Dynamoid::Config.application_timezone
-                    when :utc
-                      ActiveSupport::TimeZone['UTC'].at(value).to_datetime
-                    when :local
-                      Time.at(value).to_datetime
-                    when String
-                      ActiveSupport::TimeZone[Dynamoid::Config.application_timezone].at(value).to_datetime
-                  end
-                end
+                parse_datetime(value)
```

Comment:
It's a good refactoring. The feature could be done just with one additional line

```ruby
value = DateTime.iso8601(value).to_time.to_i if Dynamoid::Config.convert_date_to_string
```